roles/foreman_devel/tasks/rhel.yml

```diff
@@ -0,0 +1,22 @@
+---
+- name: "Check the registration"
+  shell: "subscription-manager identity > /dev/null"
+  register: sresult
+  become: true
+  ignore_errors: True
+
+- name: "Stop installation; the system is not registered"
+  fail:
+    msg: "Please register the system via subscription-manager before continuing"
+  when: "sresult.rc != 0"
+
+- name: "Enable only RHEL7 base + optional + extras repos"
```

Comment:
I think [rhsm_repository](https://docs.ansible.com/ansible/2.5/modules/rhsm_repository_module.html) is cleaner